components/FA2Transfer.tsx

```diff
@@ -116,7 +116,7 @@ const FA2Transfer = ({
   );
 
   useEffect(() => {
-    const value = (getFieldProps(makeName("token")).value as fa2Token) ?? "";
+    const value = getFieldProps(makeName("token")).value ?? "";
```

Comment:
Why end with `?? ""`? now, the type is `fa2Token | undefined`. It looks safe without `?? ""`.  If it's value is `undefined` in here, later on,  `if (!value) return;` will return. otherwise, it will continue.